src/app/components/Pagination.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="flex items-center items-center justify-center py-6">
+    <div
+      v-if="currentPage !== 1"
+      @click="changePage(currentPage - 1)"
+      class="arrow arrow-left py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+
+    <div v-for="(page, idx) in pagesArray" :key="idx">
+      <p
+        v-if="typeof page === 'number'"
+        @click="changePage(page)"
+        :class="paginationClassObject(page)"
+      >{{ page }}</p>
+      <p v-else-if="page === 'dots' && shoultRenderDots(idx)" class="py-1 px-2 mx-1">...</p>
+    </div>
+
+    <div
+      v-if="currentPage !== pagesCount"
+      @click="changePage(currentPage + 1)"
+      class="arrow arrow-right py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+export default {
+  props: {
+    limit: Number,
+    offset: Number,
+    total: Number
+  },
+
+  computed: {
+    pagesCount() {
+      return Math.floor(this.total / this.limit) + 1;
```

Comment:
Just knowledge sharing )
double Tilda could be faster. Try this one `~~(12 / 5)`
https://jsperf.com/math-floor-vs-math-round-vs-parseint/33